### DISCUSSION TITLE
How to get relative URL in fonts.css?

### DISCUSSION THREAD

**Discussion by joelstein** [spatie/laravel-google-fonts#51] [Q&A]
The fonts.css file has a `src` attribute with the absolute URL to the font files.

```
src: url(https://app.example.com/storage/fonts/9df0d02878/sinterv13ucc73fwrk3iltehus-fvqtmwcp50knma2jl7w0q5n-wu.woff2) format('woff2');
```

Is it possible to get the relative URL?

```
src: url(/storage/fonts/9df0d02878/sinterv13ucc73fwrk3iltehus-fvqtmwcp50knma2jl7w0q5n-wu.woff2) format('woff2');
```

Any ideas on how we can accomplish this?

**Comment by sebastiandedeyne** [comment]
Building the URL is handled by Laravel's filesystem configuration. You might be able to remove the prefixed env('APP_URL') from the setting to remove the domain.

https://github.com/laravel/laravel/blob/cf0b40b878b5068ed710ee39a0e1d090a1f06d60/config/filesystems.php#L42

**Comment by joelstein** [comment]
Thanks. Do you know, is there a way to inform the filesystem service not to prefix the URL in this command?

```
localizedUrl: $this->filesystem->url($this->path($url, 'fonts.css')),
```

https://github.com/spatie/laravel-google-fonts/blob/main/src/GoogleFonts.php#L111

**Reply by sebastiandedeyne** [reply]
Did you try removing the `env('APP_URL')` prefix in the filesystem config?

```diff
'public' => [
    'driver' => 'local',
    'root' => storage_path('app/public'),
-   'url' => env('APP_URL').'/storage',
+   'url' => '/storage',
    'visibility' => 'public',
    'throw' => false,
],
```

**Reply by joelstein** [reply]
I think this would have broader unintended consequences, for example, when generating URLs to assets that might be referenced outside the browser (like in an email).

Thanks, though, for your help. I was just curious is there was some way to configure this within the laravel-google-fonts library.

**Reply by sebastiandedeyne** [reply]
If you're worried about breaking other places, you could create a second version of the disk specifically for fonts.

```diff
'public' => [
    'driver' => 'local',
    'root' => storage_path('app/public'),
    'url' => env('APP_URL').'/storage',
    'visibility' => 'public',
    'throw' => false,
],
+
+'fonts' => [
+    'driver' => 'local',
+    'root' => storage_path('app/public'),
+    'url' => '/storage',
+    'visibility' => 'public',
+    'throw' => false,
+],
```

If this wouldn't work out, I'd accept a PR that extracts URL generation to it's own method so it can be easily overridden.

**Reply by joelstein** [reply]
Oh yeah, that sounds like a good idea. Let me try this and consider a PR. Thanks!

**Reply by joelstein** [reply]
That solution works perfectly. Thank you!
